core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -395,9 +393,8 @@ private void commitSimpleTransaction() {
       return;
     }
 
-    // this is always set to the latest commit attempt's snapshot id.
-    AtomicLong currentSnapshotId = new AtomicLong(-1L);
-
+    Set<Long> preTxnSnapshots =
```

Comment:
We avoid using short names that are hard to type, like `txn`. It doesn't save that much space and it makes the code harder to work with in general. I'd recommend choosing a better word instead, like `existingSnapshots` or `startingSnapshots`.